src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,230 @@
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 2: // Savings
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 3: // Client
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
+          this.chargeForm.get('penalty').disable();
+          break;
+      }
+      this.chargeForm.get('chargeCalculationType').reset();
+      this.chargeForm.get('chargeTimeType').reset();
+    });
+    this.chargeForm.get('chargeTimeType').valueChanges.subscribe((chargeTimeType) => {
+      this.chargeForm.removeControl('feeFrequency');
+      this.chargeForm.removeControl('feeInterval');
+      this.chargeForm.removeControl('feeOnMonthDay');
+      this.chargeForm.removeControl('addFeeFrequency');
+      if (this.chargeForm.get('chargeAppliesTo').value !== 4) {
+        this.chargeForm.get('penalty').enable();
+      }
+      switch (chargeTimeType) {
+        case 6: // Annual Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl('', Validators.required));
+          break;
+        case 7: // Monthly Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl(''));
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.min(1), Validators.max(12), Validators.pattern('^[1-9]\\d*$')]));
+          this.repeatEveryLabel = 'Months';
+          break;
+        case 9: // Overdue Fee
+          this.chargeForm.get('penalty').setValue(true);
+          this.chargeForm.get('penalty').disable();
+          this.chargeForm.addControl('addFeeFrequency', new FormControl(false));
+          this.chargeForm.get('addFeeFrequency').valueChanges.subscribe((addFeeFrequency) => {
+            if (addFeeFrequency) {
+              this.chargeForm.addControl('feeFrequency', new FormControl('', Validators.required));
+              this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^[1-9]\\d*$')]));
+            } else {
+              this.chargeForm.removeControl('feeFrequency');
+              this.chargeForm.removeControl('feeInterval');
+            }
+          });
+          break;
+        case 11: // Weekly Fee
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^[1-9]\\d*$')]));
+          this.repeatEveryLabel = 'Weeks';
+          break;
+      }
+    });
+    this.chargeForm.get('currencyCode').valueChanges.subscribe((currencyCode) => {
```

Comment:
I went through the component again to check through the final validations and what not. I've discovered that the amount should be under a certain decimal place (which is different for each currency), or else it simply would not make sense in terms of accounting sense etc. This is not implemented in the community app, which in the current version only accepts integers (this is just wrong, even in the API, they have plenty of examples with decimal amounts), but I think that it is quite a necessary validation to put in place.

Reply:
I really appreciate you taking so much interest and putting effort into this!! Good. 👍